File: pi_integration/src/pi_integration_serial.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main (int argc, char *argv[])
{
  int i,num_steps;
  double x, sum, step, pi;
  clock_t t_start, t_end;
  
  num_steps=500000000;
  x=0;
  sum = 0.0;
  step = 1.0/(double) num_steps;
  t_start = (double) clock();
  for (i=0; i < num_steps; i++)
    {
      x = (i+0.5)*step;
      sum = sum + 4.0/(1.0+x*x);
    }
  
  pi = step * sum;
  t_end = (double) clock();
  printf("Value of pi = %g\n",pi);
  printf("Expended wall clock time = %.20f\n", (double)(t_end - t_start)/CLOCKS_PER_SEC);
  return EXIT_SUCCESS;
}

File: pi_integration/src/pi_integration_serial.f90
PROGRAM pi_integration_serial
  IMPLICIT NONE
  INTEGER :: i
  REAL(kind=8) :: x, sum, step, pi
  REAL(kind=8) :: t_start, t_end

  INTEGER(kind=8), PARAMETER :: num_steps = 500000000

  sum=0.0
  step = 1.0/num_steps

  CALL cpu_time(t_start)

  DO i=0,num_steps
     x = (i+0.5)*step
     sum = sum + 4.0/(1.0+x*x)
  END DO

  pi = step * sum
  
  CALL cpu_time(t_end)

  WRITE(*,*) "Value of pi = ",pi
  WRITE(*,*) "Expended wall clock time = ",t_end-t_start
  
END PROGRAM pi_integration_serial
